src/Icicle/Sea/FromAvalanche/Prim.hs

```diff
@@ -165,6 +165,8 @@ seaOfPrimArray p
       -> PDAlloc (prefixOfValType (ArrayT t) <> "put_immutable") Nothing
      PrimArraySwap t
       -> PDFun   (prefixOfValType (ArrayT t) <> "swap") Nothing
+     PrimArrayDel t
+      -> PDFun   (prefixOfValType (ArrayT t) <> "delete") Nothing
```

Comment:
hm, I would prefer `put_immutable` to be the default like it currently is though. Just in case the mutation recovery is incorrect. If `mutable` is the default and we try to find places where we should not mutate, it's a bit more scary.